packages/listbox/src/ListboxMixin.js

```diff
@@ -670,7 +670,9 @@ const ListboxMixinImplementation = superclass =>
         ev.stopPropagation();
       }
       this.__onChildCheckedChanged(ev);
-      this.requestUpdate('modelValue', this.modelValue);
+
+      // don't send this.modelValue as oldValue, since it will take modelValue getter which takes it from child elements, which is already the updated value
```

Comment:
We need to pass along the old value (it's a best practice and we should be able to detect)

@palash2601  did this as a merge request recently. It prevented unintended behavior inside the combobox textbox syncing.
@palash2601 could you elaborate on this an would it be possible to add a unit test for this?

Also I think if we want to do it correctly, I was thinking, we should do:

```js
setCheckedIndex(...) {
const oldModelValue = this.multipleChoice ? [...this.modelValue] : this.modelValue; 
... // body of method
this.requestUpdate('modelValue', oldModelValue);
}
```

And then we should add a `hasChanged` method to the static prop getter as well that is able to do a deep equal for arrays.

Also, we need to make sure all 'checked changed logic' is handled via this method (everything is now, except when checked events are fired by LionOptions).

Reply:
Hey @tlouisse , @jorenbroekema ,

This change earlier was done for [https://github.com/ing-bank/lion/issues/993](issue). More detailed RCA is in comments of this issue

Reply:
```js
const oldModelValue = this.multipleChoice ? [...this.modelValue] : this.modelValue; 
```

Won't work. It goes through the modelValue getter which fetches from the lion-option's --> checked. This is already "updated" so you don't actually get the old model value, you get the new one. And then if you pass that to `this.requestUpdate` the update never happens because the diff does not pass.

Reply:
```js
  __proxyChildModelValueChanged(ev) {
      // We need to redispatch the model-value-changed event on 'this', so it will
      // align with FormControl.__repropagateChildrenValues method. Also, this makes
      // it act like a portal, in case the listbox is put in a modal overlay on body level.
      if (ev.stopPropagation) {
        ev.stopPropagation();
      }
      this.__onChildCheckedChanged(ev);

      // don't send this.modelValue as oldValue, since it will take modelValue getter which takes it from child elements, which is already the updated value
      this.requestUpdate('modelValue', this.__oldModelValue);
      this.dispatchEvent(
        new CustomEvent('model-value-changed', { detail: { element: ev.target } }),
      );
      this.__oldModelValue = this.modelValue;
    }
```

This seems to work, now it passes the diff check and I actually get the updated callback for modelValue when I do backspace on a preselected option.

Reply:
> ```js
> const oldModelValue = this.multipleChoice ? [...this.modelValue] : this.modelValue; 
> ```
> 
> Won't work. It goes through the modelValue getter (ChoiceGroupMixin) which fetches from the lion-option's --> checked. This is already "updated" so you don't actually get the old model value, you get the new one. And then if you pass that to `this.requestUpdate` the update never happens because the diff does not pass.

If I'm right you only get the new one if 'checked' was set on an individual option, right? (because it's already updated then by the child itself)
In all other cases, it's the body of `setCheckedIndex` that does the change. 
Ideally, all checked states would be managed by the parent via this method and a child would only request a checked state change.

This would be some work to implement, so maintaining a `this.__oldModelValue` would also do the job.